gate-plugins/src/main/kotlin/com/netflix/spinnaker/gate/plugins/deck/DeckPluginCache.kt

```diff
@@ -38,7 +39,8 @@ class DeckPluginCache(
   private val springPluginStatusProvider: SpringPluginStatusProvider,
   private val pluginInfoReleaseProvider: PluginInfoReleaseProvider,
   private val registry: Registry,
-  private val springStrictPluginLoaderStatusProvider: SpringStrictPluginLoaderStatusProvider
+  private val springStrictPluginLoaderStatusProvider: SpringStrictPluginLoaderStatusProvider,
+  private val pluginsCacheDirectory: String?
```

Comment:
Instead of nullable, thoughts on wrapping this in `Optional`?

Reply:
I think, it would help to be more readable and keep the same functionality. I will change it